scalableminds/webknossos
# Merge editable mappings

- Adds merge functionality for editable mapping (a.k.a. supervoxel proofreading) volume layers
- Fixes a bug in duplicating such annotations
- Adds small refactoring `Instant.since`
- Changes the merge modal from a checkbox to individual buttons to emphasize the difference between the options:
![image](https://user-images.githubusercontent.com/1390035/236186475-3ac3913a-3a8d-4589-bc0f-ebd3c556d608.png)


### TODO
 - [x] Depends on https://github.com/scalableminds/fossildb/pull/38 – after that release, change fossilDB version here and update migration guide

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- While the new fossilDB is not released, this can be tested by checking out the FossilDB branch `fix-get-multiple-keys-pagination`, running `sbt "run --columnFamilies skeletons,skeletonUpdates,volumeData,volumes,volumeUpdates,editableMappings,editableMappingUpdates,editableMappingsInfo,editableMappingsAgglomerateToGraph,editableMappingsSegmentToAgglomerate"` there and `yarn start noFossilDB` in webknossos.
- Create two explorative editable mapping annotations on the same dataset, merge/split some agglomerates
- Use the merge modal to merge one with the other
- Result should have updates of both mappings applied
- Merge modal should be more understandable as well (feedback welcome)

### Issues:
- fixes #6955 
- fixes #7015

------
